E3Next/Classes/Magician.cs

```diff
@@ -197,407 +308,386 @@ public static void ArmPets()
+                    PickUpItem(item);
+                    if (item == MQ.Query<string>($"${{Cursor.Name}}"))
                     {
-                        e3util.ClearCursor();
+                        MQ.Cmd("/destroy");
                     }
+                    if (E3.CharacterSettings.AutoPetDebug) E3.Bots.Broadcast($"\amDebug: CU: Item: {item} was destroyed.");
                 }
-
-                return (true, null);
-            }
-            else
-            {
-                return (false, $"Unable to cast {itemToSummon} because it wasn't ready");
             }
+
+            if (E3.CharacterSettings.AutoPetDebug) E3.Bots.Broadcast($"\acDebug: LazCleanUp process Finished.");
         }
 
-        private static bool CheckInventory()
+        /// <summary>
+        /// Keeps an inventory slot open for summoned shit.
+        /// </summary>
+        [ClassInvoke(Data.Class.Magician)]
+        public static void KeepOpenInvSlot()
         {
-            // clean up any leftovers
-            var bag = _armorOrHeirloomBag;
-            while (MQ.Query<int>($"${{FindItemCount[={bag}]}}") > 0)
+            if (!E3.CharacterSettings.KeepOpenInventorySlot) return;
+            if (Basics.InCombat()) return;
+            if (!e3util.ShouldCheck(ref _nextInventoryCheck, _nextInventoryCheckInterval)) return;
+
+            var slotToKeepOpen = "pack10";
+
+            // if we have no open inventory slots, return
+            var freeInv = MQ.Query<int>("${Me.FreeInventory}");
+            if (freeInv == 0)
             {
-                if (!DestroyIfEmpty(bag)) return false;
+                if (E3.CharacterSettings.AutoPetWeapons)
+                {
+                    E3.Bots.Broadcast("\arError: No free inventory space and auto pet weapons is on - toggling off so inventory space can be freed up");
+                    E3.CharacterSettings.AutoPetWeapons = false;
+                }
+
+                return;
             }
 
-            bag = "Huge Disenchanted Backpack";
-            while (MQ.Query<int>($"${{FindItemCount[={bag}]}}") > 0)
+            // check if there's anything there
+            var slotQueryResult = MQ.Query<string>($"${{Me.Inventory[{slotToKeepOpen}]}}");
+            if (slotQueryResult == "NULL") return;
+
+            // find a spot to move it to
+            var containerWithOpenSpace = 0;
+            for (int i = 1; i <= 9; i++)
             {
-                if (!DestroyIfEmpty(bag)) return false;
+                var containerSlots = MQ.Query<int>($"${{Me.Inventory[pack{i}].Container}}");
+                if (containerSlots == 0) continue;
+
+                var containerItemCount = Math.Abs(MQ.Query<int>($"${{InvSlot[pack{i}].Item.Items}}"));
+                if (containerItemCount < containerSlots)
+                {
+                    containerWithOpenSpace = i;
+                    break;
+                }
             }
 
-            bool DestroyIfEmpty(string containerName)
```

Comment:
why was this method removed? this was there to protect against destroying bags that were left around and got other items put into them